contracts/StealthVault.sol

```diff
@@ -98,16 +99,22 @@ contract StealthVault is Governable, CollectableDust, ReentrancyGuard, IStealthV
     bonded[governor] = bonded[governor] + (_penalty - _amountReward);
   }
 
+  modifier onlyEOA() {
+    uint256 _gasLeftPlusBuffer = gasleft() + gasBuffer;
+    require(_gasLeftPlusBuffer >= (block.gaslimit * 63) / 64, 'SV: eoa gas check failed');
+    _;
+  }
+
   // Hash
   function validateHash(
     address _caller,
     bytes32 _hash,
     uint256 _penalty
-  ) external override nonReentrant() returns (bool) {
+  ) external virtual override onlyEOA() nonReentrant() returns (bool _valid) {
```

Comment:
Im confused, making this onlyEOA wouldnt that mean that you can only call it from an EOA, and you would never be able to validate from a job ? 
Maybe the name confuses me, and it should be something like `onlyOneHop` or something like that. Since ... I think we are lying here, right ?

Reply:
yeah, naming is a bit confusing here, will change to (`oneCallStack` ?), other check is for `_caller` which is the "user/EOA".